src/main/java/me/arminb/sara/dao/UserDAOImpl.java

```diff
@@ -0,0 +1,125 @@
+package me.arminb.sara.dao;
+
+import com.mongodb.*;
+import com.mongodb.client.MongoCollection;
+import com.mongodb.client.MongoCursor;
+import com.mongodb.client.MongoDatabase;
+import com.mongodb.client.result.DeleteResult;
+import com.mongodb.client.result.UpdateResult;
+import com.mongodb.util.JSON;
+import me.arminb.sara.entities.User;
+import org.bson.Document;
+import org.bson.types.ObjectId;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import java.util.ArrayList;
+import java.util.List;
+
+@Repository("userDAO")
+public class UserDAOImpl implements UserDAO {
+
+    Logger logger = LoggerFactory.getLogger(UserDAOImpl.class);
+
+    @Autowired
+    private MongoDatabase database;
+
+    @Override
+    public List<User> findAll() {
+        List<User> list = new ArrayList();
+        try {
+            MongoCollection<Document> collection = database.getCollection("users");
+            MongoCursor<Document> iterator = collection.find().iterator();
+            while (iterator.hasNext()) {
+                Document doc = iterator.next();
+                User user = new User();
+                user.setId((ObjectId)doc.get("_id"));
+                user.setUsername(doc.get("user_name").toString());
+                user.setEmail(doc.get("user_email").toString());
+                user.setPassword(doc.get("user_password").toString());
+                list.add(user);
+            }
+        } catch (MongoException e) {
+            logger.info("Failed to select users due to"+ e);
+        } finally {
+            return list;
```

Comment:
I wouldn't make the decision to send the empty list up to the caller method here because then the caller method cannot distinguish between the situation where there no user in the system and when an error has happened. I would translate this exception to something like DataAccessException and throw it up to be catch by caller methods.

You need to define the new exception class like this 

class DataAccessException extends Exception {
}

then you can throw the exception after logging appropriate error or warn (info is not a wise choice here) message using the following command:
throw new DataAccessException();

this is called exception translation. basically giving the exception specific to the mongo layer, you translate it to an exception which is specific to your layer (DAO in this case). In this way the Service layer can make sense out of DAO layer without knowing about the existence of the mongo layer.